sanctuary fixes

OmarTawfik opened the pull request:
- reset git state before running tests
- fix a bunch of nits with string/error reporting in CI
- trim null terminators from input files

Xanewok commented on crates/solidity/testing/sanctuary/src/tests.rs:
```diff
@@ -73,6 +73,8 @@ pub fn run_test(file: &SourceFile, events: &Events) -> Result<()> {
     let source = file
         .path
         .read_to_string()?
+        // Some files are null terminated, remove the null character:
```
How are handling it right now? Is the parser not resilient enough to handle it?

OmarTawfik replied:
No, because it is not part of our grammar, and is not supported by `solc`.
I suspect it is an issue with how `sanctuary` stores files. I filed an issue here and added it as a comment:

https://github.com/tintinweb/smart-contract-sanctuary/issues/30

Xanewok commented on crates/solidity/testing/sanctuary/src/reporting.rs:
```diff
@@ -67,9 +66,8 @@ impl Reporter {
-        std::io::stdout()
-            .write_all(&buffer.contents_formatted()[..])
-            .unwrap();
+        let contents = String::from_utf8(buffer.contents_formatted()).unwrap();
```
Do we want to assert that the output is UTF-8? What's the reason behind that?

OmarTawfik replied:
It is expected to be valid UTF8. I split this to make it a bit clear.

Xanewok commented on crates/solidity/testing/sanctuary/src/datasets.rs:
```diff
@@ -61,6 +61,12 @@ impl DataSet {
     }
 
     pub fn checkout_directory(&self, directory: &str) -> Result<&Vec<SourceFile>> {
+        Command::new("git")
```
Do we make the directory dirty as part of the sanctuary run? If so, should we also run a `git clean` as well?

OmarTawfik replied:
Nop. But for local development, I often ended up needing to edit these files in place, to experiment with our error reporting/locations.
I suspect others will be doing the same as well, so I added it for extra insurance. I will add a comment 👍

Xanewok reviewed (commented):
It'd be great to have more context and reason behind these changes :pray:

OmarTawfik commented:
> It'd be great to have more context and reason behind these changes 🙏

@Xanewok apologies. I added it to the [commit message](https://github.com/NomicFoundation/slang/pull/855/commits/162ad7bf0e1af9378cd89f15b610c4bf0fe4c350), but for some reason, it wasn't copied to the PR. Added!

Xanewok reviewed (approved):
Thanks for the comments and the links, these are helpful!
